### PR TITLE
Update AzureSearchAdminKey detector

### PR DESCRIPTION
**Pull request by rgmz**
### Description:
There are a number of problems with this detector. This Pr attempts to fix some of the glaring issues.

e.g.,
>  Verification issue: unexpected HTTP response status 400 | {"error":{"code":"","message":"Service name must only contain lowercase letters, digits or dashes, cannot start or end with or contain consecutive dashes and is limited to 60 characters."}}

### Checklist:
* [x] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/azuresearchadminkey/azuresearchadminkey.go [hunk 1]
```diff
@@ -3,10 +3,9 @@ package azuresearchadminkey
 import (
 	"context"
 	"fmt"
-	"net/http"
-	"strings"
-
 	regexp "github.com/wasilibs/go-re2"
+	"io"
```
Comment (ahrav) [inline pkg/detectors/azuresearchadminkey/azuresearchadminkey.go:7]: nit: Could we please get these import sorted.

```
import (
	"context"
	"fmt"
	"io"
	"net/http"

	regexp "github.com/wasilibs/go-re2"

	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
)
```

File: pkg/detectors/azuresearchadminkey/azuresearchadminkey.go [hunk 2]
```diff
@@ -23,79 +22,95 @@ var _ detectors.Detector = (*Scanner)(nil)
...
 
-			s1 := detectors.Result{
+	for key := range keyMatches {
+		for service := range serviceMatches {
+			r := detectors.Result{
 				DetectorType: detectorspb.DetectorType_AzureSearchAdminKey,
-				Raw:          []byte(resMatch),
-				RawV2:        []byte(resMatch + resServiceMatch),
+				Raw:          []byte(key),
+				RawV2:        []byte(`{"service":"` + service + `","key":"` + key + `"}`),
```
Comment (ahrav) [inline pkg/detectors/azuresearchadminkey/azuresearchadminkey.go:67]: Changing the `RawV2` value will break some enterprise customers, as we currently have secrets verified using the old `key+service` format. This change would orphan previously verified credentials and create duplicates. Is there a reason we can't continue supporting the old format?

**Review by ahrav — CHANGES_REQUESTED** [review]
Single blocking change as it pertains to reformatting `RawV2`, otherwise LGTM!

File: pkg/detectors/azuresearchadminkey/azuresearchadminkey.go (hunk 2 above)
Comment (rgmz) [reply pkg/detectors/azuresearchadminkey/azuresearchadminkey.go:67]: > Is there a reason we can't continue supporting the old format?

Obviously you get the final say here, but #2128 has been a PITA for me and others using OSS.

TL;DR Most of the existing `RawV2` values are suboptional because they smash together text (sometimes without a delimiter) and are devoid of context/keywords. This makes it impossible to programmatically re-construct and re-verify results.

Comment (ahrav) [reply pkg/detectors/azuresearchadminkey/azuresearchadminkey.go:68]: My guess is that when `RawV2` was proposed, we didn't account for its visibility in OSS reporting. Wonder if we can find a solution that preserves readability for OSS while avoiding duplicates on the enterprise side?

This likely requires a broader discussion, and I think @zricethezav  will have the most input, but I agree that combining multiple credentials makes them difficult to interpret and largely ineffective

**Comment by rgmz** [comment]
Seems like a new/different error introduced by the azure storage change. I'll take a look when I'm back on the computer.

File: pkg/detectors/azuresearchadminkey/azuresearchadminkey.go (hunk 2 above)
Comment (rgmz) [reply pkg/detectors/azuresearchadminkey/azuresearchadminkey.go:68]: > My guess is that when RawV2 was proposed, we didn't account for its visibility in OSS reporting. Wonder if we can find a solution that preserves readability for OSS while avoiding duplicates on the enterprise side?

I can't speak to the impact on enterprise. I can revert the change to `RawV2`, though I'd prefer to improve it (eventually).
